pkg/agent/openflow/pipeline.go

```diff
@@ -289,7 +289,7 @@ var (
 
 	globalVirtualMAC, _ = net.ParseMAC("aa:bb:cc:dd:ee:ff")
 	hairpinIP           = net.ParseIP("169.254.169.252").To4()
-	hairpinIPv6         = net.ParseIP("FE80::aabb:ccdd:eeff").To16()
+	hairpinIPv6         = net.ParseIP("ee80::aabb:ccdd:eeff").To16()
```

Comment:
ee80: is not a reserved IPv6 address, right? I think we should use a reserved address to avoid collision with real world IP.
I know "FE80:" will cause problems as the container will have an address in that subnet allocated automatically, but we should still find a reserved IP here.

Reply:
No, not a reserved one.
I agree that a reserved one is the best choice but I failed to find one here. https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml
Considering the large number of IPv6 addresses, address like "ee80" might be safe enough to use?

Reply:
I haven't thought about the impact if a collision really happens. But it doesn't sound right to me to use a real world IP to represent harpin IP. I think there should be similar cases in various networking solutions and there should be some common virtual IP to use.
cc @jianjuns @antoninbas to see if they have any idea.

Reply:
@salv-orlando might have some suggestions here.

Reply:
BTW, I would not add bug fixes like this in an e2e test PR. It is hard to track from the commit history.

Reply:
I found the bug when enabling e2e tests. So I think these tests help verify the fix?

Reply:
Maybe we should use a unique local address here [rfc4193](https://tools.ietf.org/html/rfc4193).

Reply:
I think it works. Thank you!

Reply:
For IPv4 we use a link local address, and therefore I believe that link local should be used for IPv6 as well.
IPv6 stacks are commonly using the MAC to generate the link local address, e.g for MAC 00:50:56:a6:f2:f5 the LL IPv6 address fe80::250:56ff:fea6:f2f5.

Reply:
It is kind different in IPv6 case. As mentioned by Quan, an automatically generated IPv6 link local address will be allocated to a device. Then, the device considers itself in the same subnet as our IPv6 link local address. As a result, traffic fails. It has been verified by TestProxyServiceSessionAffinity and TestProxyHairpin testcases.
So we need a reserved IP address also it cannot be a link local one. A unique local address should work.

Reply:
I get your point. Thanks for elaborating!

Reply:
@lzhecheng , this patch title is a little bit misleading as it's not only enable the test but also fix the IPv6 bugs. We can change the patch title to reflect the real problems we are resolving.

Reply:
Good idea. Updated PR title and patch description.